website/project/model.py

```diff
@@ -1604,7 +1604,7 @@ def update(self, fields, auth=None, save=True):
                     {
                         'id': value.get('id'),
                         'year': value.get('year'),
-                        'copyrightHolders': value.get('copyrightHolders')
+                        'copyrightHolders': value.get('copyrightHolders') or value.get('copyright_holders', [])
```

Comment:
The APIv1 (current OSF front-end) update returns `copyright_holders` from [`licensePicker.js`](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/static/js/licensePicker.js#L219) whereas APIv2 will directly return `copyrightHolders` [here](https://github.com/CenterForOpenScience/osf.io/blob/8cb2fb682504c488232d7117645811ba3ce9fe97/api/nodes/serializers.py#L103).

APIv2 returns `copyrightHolders` because in [`set_license()`](https://github.com/CenterForOpenScience/osf.io/blob/8cb2fb682504c488232d7117645811ba3ce9fe97/website/project/licenses/__init__.py#L164) the license details are checked against the required properties of the license (which is literally `NodeLicense.properties.copyrightHolders`). So, the key passed to `set_license()` needs to end up being `copyrightHolders`.

Alternatively, I could change the line in licensePicker.js to return `copyrightHolders` in its payload.